tests/src/test/scala/com/nvidia/spark/rapids/CollectionOpSuite.scala

```diff
@@ -19,22 +19,18 @@ package com.nvidia.spark.rapids
 import org.apache.spark.sql.functions.map_concat
 
 class CollectionOpSuite extends SparkQueryCompareTestSuite {
-  testGpuFallback(
-    "MapConcat with Array keys fall back",
-    "ProjectExec",
-    ArrayKeyMapDF, 
-    execsAllowedNonGpu = Seq("ProjectExec", "ShuffleExchangeExec")) {
+  testSparkResultsAreEqual(
+    "MapConcat with Array keys",
+    ArrayKeyMapDF) {
```

Comment:
Can we add in some integration tests as well or instead of these? It would also be good to try and add/update the data gen for the tests so it is much more likely to have collisions. This is especially true for nested types like arrays or structs where the probability of a collision goes way down.

Reply:
> Can we add in some integration tests as well or instead of these? It would also be good to try and add/update the data gen for the tests so it is much more likely to have collisions.

If you mean adding tests in Python Integration test, unfortunately, it is difficult. Because `List` or `Struct` is not hashable in Python.

Reply:
```python
>>> schema = StructType([
...     StructField('name', StringType(), True),
...     StructField('properties', MapType(ArrayType(StringType()),StringType()),True)])
>>> dataDictionary = [
...         ('James',{['hair', 'hair']:'black',['eye', 'eye']:'brown'})]
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
TypeError: unhashable type: 'list'
```

Reply:
@revans2 is there any workaround way that can generate random `MapType` data in IT?

Reply:
Hi @revans2, I find it is a little difficult to update the `MapGen` to support the nested keys because currently the `MapGen` uses `dict` however, `list` and `struct` are unhashable in dict.

A workaround way is firstly generating lists of structs using `ListGen` and `StructGen` and then convert the data to map type. But I am not sure whether this is a good workaround.

Reply:
That is fine. Sorry I didn't respond sooner. I have been stuck on something else.